config/openshift/v4_8_exp/translate_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2021 Red Hat, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.)
+
+package v4_8_exp
+
+import (
+	"testing"
+
+	"github.com/coreos/fcct/config/common"
+
+	"github.com/coreos/vcontext/path"
+	"github.com/coreos/vcontext/report"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestElidedFieldWarning tests that we warn when transpiling fields to an
+// Ignition config that can't be represented in an Ignition config.
+func TestElidedFieldWarning(t *testing.T) {
+	in := Config{
+		Metadata: Metadata{
+			Name: "z",
```

Comment:
It feels a bit awkward that `metadata.name` is a required field but is elided and not warned about when rendering raw Ignition. Although I'm not sure if we could get around the circular dependency issue if we wanted to check the status of `raw` so I'm okay just living with it.

Reply:
I intentionally didn't warn on the `metadata` fields, on the theory that they don't affect how the machine is configured.  The `openshift` fields do affect the machine, so it could be surprising if those changes are not applied.

Reply:
That's fair, although I couldn't say with certainty that there isn't something potentially keying off a metadata field that might trigger machine-level modifications.

Reply:
I can't either.  But if we warn on `metadata`, the `--raw` flag will always emit warnings and most of the time I suspect they wouldn't be meaningful.  So there's a risk of warning fatigue.